File: python/scriptspython/repeticao/fatorial.js
function fatorial(numeroDeEntrada) {
  let soma = 1;
  if (numeroDeEntrada <= 0 || numeroDeEntrada == 1) {
    return console.log("Insira um numero positivo ou maior que 1");
  } else {
    while (numeroDeEntrada >= 1) {
      soma *= numeroDeEntrada;
      numeroDeEntrada -= 1;
    }
  }
  return soma;
}

let numeroDeEntrada = 4;
let resultado = fatorial(numeroDeEntrada);

console.log(`O fatorial de ${numeroDeEntrada} é ${resultado}`);

File: python/exercicios/scriptspython/repeticao/fatorial.py


def calcularFatorial(numeroDeEntrada):
       soma = 1
       if numeroDeEntrada <= 0 or numeroDeEntrada == 1:
              print("Apenas numeros positivos ou maior que 1")
       else:
              while numeroDeEntrada >= 1:
                     soma *= numeroDeEntrada 
                     numeroDeEntrada -= 1
       return soma

numeroDeEntrada = int(input("Digite um numero de entrada \n"))
fatorial = calcularFatorial(numeroDeEntrada)
print(f"O fatorial do numero {numeroDeEntrada} é {fatorial}")